libsolidity/ast/Types.cpp

```diff
@@ -40,10 +40,35 @@
 
 #include <limits>
 
+#include <boost/multiprecision/cpp_dec_float.hpp>
+
 using namespace std;
 using namespace dev;
 using namespace dev::solidity;
 
+namespace
+{
+
+bool fitsPrecision(bigint const _base, bigint const _exp)
+{
+	using boost::multiprecision::log10;
+	using boost::multiprecision::ceil;
+	using bigfloat = boost::multiprecision::number<boost::multiprecision::cpp_dec_float<50>>;
```

Comment:
Does that mean it allows up to 50 decimal digits? What happens if `_base + 1` is larger than 50 decimal digits?

According to boost these are also defined:
```
typedef number<cpp_dec_float<50> > cpp_dec_float_50;
typedef number<cpp_dec_float<100> > cpp_dec_float_100;
```

Reply:
@axic The precision will be reduced after creating ``cpp_dec_float_50`` with ``base + 1``. That means a if ``base`` has more than 50 digits, any additional digit will be represented with a zero. This does not affect the size of it.